### PR TITLE
feat(structures): Embed interface

### PR DESCRIPTION
**Pull request by matthew119427**
This PR adds an `index.ts` file as well as the `Embed` interface structure.

### CODE REVIEW

**Comment by bsian03** [comment]
You seemed to have made the commit from your beta branch on your fork instead of the base repository, can you reset and cherry-pick the user interface commit? Same for #25

**Comment by Khaaz** [comment]
lgtm, just waiting for the cherry pick (conflict / extend) with the Base

File: structures/Base.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  public id: string;
+
+  constructor(id: string) {
+    this.id = id;
+  }
+
+  toString() {
+    return `[${this.constructor.name}${this.id ? ` ${this.id}` : ''}]`;
+  }
+}
```
Comment (bsian03) [inline structures/Base.ts:11]: Won't need this for now

File: structures/Embed.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+export interface EmbedAuthor {
+  icon_url?: string;
+  name?: string;
+  proxy_icon_url?: string;
+  url?: string;
+}
+
+export interface EmbedFooter {
```
Comment (bsian03) [inline structures/Embed.ts:8]: Missing proxy url

File: structures/Embed.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  color?: number;
+  description?: string;
+  fields?: EmbedField[];
+  footer?: EmbedFooter;
+  image?: EmbedMedia;
+  provider?: EmbedProvider;
+  thumbnail?: EmbedMedia;
+  timestamp?: Date;
+  title?: string;
+  type?: 'rich' | 'image' | 'video' | 'gifv' | 'article' | 'link';
```
Comment (bsian03) [inline structures/Embed.ts:47]: Can this be type aliased

**Review by bsian03 — CHANGES_REQUESTED** [review]
Can the interfaces be in alphabetical order as well?

**Comment by SiroTM** [comment]
EmbedFooter is defined twice

File: structures/Embed.ts (hunk 3 above)
Comment (Khaaz) [reply structures/Embed.ts:47]: an enum for this would be good yes

**Comment by matthew119427** [comment]
See #42

**Pull request closed without merging**